src/Microsoft.Performance.Toolkit.Engine/EngineCreateInfo.cs

```diff
@@ -71,5 +72,17 @@ public string ExtensionDirectory
         ///     advanced scenarios.
         /// </summary>
         public IAssemblyLoader AssemblyLoader{ get; set;  }
+
+        /// <summary>
+        ///     Gets or sets the <see cref="VersionChecker"/> to
+        ///     use for loading plugins. This property may be <c>null</c>.
+        ///     A <c>null</c> value indicates to use the default loading
+        ///     behavior.
+        ///     <para/>
+        ///     The vast majority of use cases will not need to use
+        ///     this property. Changing the version behavior is for
+        ///     advanced scenarios.
+        /// </summary>
+        public VersionChecker Versioning { get; set; }
```

Comment:
Do we really want this to be public? Why would someone ever want to change this? If this is for testing, I would say make it protected and set tests to see it